starter_code/src/movies.js

```diff
@@ -1,19 +1,138 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
+
+function ratesAverage(arr) {
+    const avg =
+        arr.reduce(function (accumulator, movie) {
+            return accumulator + parseFloat(movie.rate);
+        }, 0) / arr.length;
+    return parseFloat(avg.toFixed(2));
+}
 
- 
 // Iteration 2: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(arr) {
+    const dramaFilms = arr.filter(function (movie) {
+        if (movie.genre.includes("Drama")) {
+            return true;
+        }
+    });
+
+    if (dramaFilms.length === 0) {
+        return 0;
+    }
+
+    const dramaAvg =
+        dramaFilms.reduce(function (total, movie) {
+            return total + parseFloat(movie.rate || 0);
+        }, 0) / dramaFilms.length;
+    return parseFloat(dramaAvg.toFixed(2));
+}
 
 // Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
 
+function orderByYear(arr) {
```

Comment:
In this particular function, for future reference, I would try and create a copy of the original array. the sort function mutates the original array. In this case there's no issue at all, but further along the line in the bootcamp (and beyond) people tend not to mutate the original value.

Still, the logic is 100% correct :)